src/ControllerStatusIO.c

```diff
@@ -99,7 +99,14 @@ BOOL Ros_Controller_Initialize()
                 {
                     for (int jointIndex = 0; jointIndex < g_Ros_Controller.ctrlGroups[groupIndex]->numAxes; jointIndex += 1)
                     {
-                        sprintf(g_nodeConfigSettings.joint_names[(groupIndex * MP_GRP_AXES_NUM) + jointIndex], DEFAULT_JOINT_NAME_FMT, groupIndex + 1, jointIndex + 1);
+                        if (g_Ros_Controller.numGroup == 1)
+                        {
+                            sprintf(g_nodeConfigSettings.joint_names[(groupIndex * MP_GRP_AXES_NUM) + jointIndex], DEFAULT_JOINT_NAME_FMT_SINGLE, jointIndex + 1);
+                        }
+                        else
+                        {
+                            sprintf(g_nodeConfigSettings.joint_names[(groupIndex * MP_GRP_AXES_NUM) + jointIndex], DEFAULT_JOINT_NAME_FMT_MULTI, groupIndex + 1, jointIndex + 1);
+                        }
```

Comment:
what about using the `if` to select the appropriate format constant? Would result in less duplicated code.

Reply:
The format constants have different number of format specifiers/`sprintf`s have different numbers of arguments (joint vs group + joint). So shouldn't it stay separate?

Reply:
Ah, yes, you're right. That's unfortunate.

*Technically*, we could just always pass `groupIndex` and `jointIndex` to `sprintf(..)`, even if one of the formats only uses the first. According to the C standard, excess arguments should be ignored.

But we probably don't want to / can't rely on that.